src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/processor/EnvelopeProcessor.java

```diff
@@ -71,4 +77,15 @@ public void markAsUploadFailed(String reason, Envelope envelope, String containe
                 .build()
         );
     }
+
+    public void markAsGenericFailure(String reason, Envelope envelope, String container, String zipFileName) {
+        envelopeStateRepository.save(
+            EnvelopeStatusBuilder
+                .newEnvelopeStatus(container, zipFileName)
+                .withEnvelope(envelope)
+                .withStatus(DOC_FAILURE)
+                .withReason(reason)
+                .build()
+        );
+    }
```

Comment:
Looks like we've got 3 almost identical methods now:
- `markAsUploaded`
- `markAsUploadFailed`
- `markAsGenericFailure`

Perhaps the common part could be extracted to a private method?

Reply:
All generalisation can be done once we have a bigger scope of what's coming where from. Premature refactor is the root of evil. I am fully aware of similarity